graphql_compiler/schema_merging/utils.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+
+from graphql.language.visitor import Visitor, visit
+
+
+class SchemaError(Exception):
+    pass
+
+
+SchemaData = namedtuple('SchemaData', ['query_type', 'scalars', 'directives', 'has_extension'])
+
+
+class GetSchemaDataVisitor(Visitor):
+    """Gather information about the schema to aid any transforms."""
+    def __init__(self):
+        self.query_type = None  # str, name of the query type, e.g. 'RootSchemaQuery'
+        self.scalars = set()  # Set[str], names of scalar types
+        self.directives = set()  # Set[str], names of directives
+        self.has_extension = False
+        self.schema_data = None
+
+    def enter_TypeExtensionDefinition(self, node, *args):
```

Comment:
What is a TypeExtension? I have never seen it.

Reply:
Something like `extend RootSchemaQuery { something: something }`, some of these exist in the extension schema in Predrag's demo. They're not expected to exist in normal schemas I don't think.